src/codegen.cpp

```diff
@@ -92,6 +92,49 @@
 
 using namespace llvm;
 
+//Drag some useful type functions into our namespace
+//to reduce verbosity of our code
+auto getInt1Ty(LLVMContext &ctxt) {
+    return Type::getInt1Ty(ctxt);
```

Comment:
Would it be too clever to write this as:
```suggestion
template <class T> // covers Module, IRBuilder, Type, Value, etc.
LLVMContext &getContext(T &m) {
    return m->getContext();
}
LLVMContext &getContext(LLVMContext &Ctx) {
    return Ctx;
}

template <class T>
auto getInt1Ty(T &ctx) {
    return Type::getInt1Ty(getContext(ctx));
```

and later:
```c++
LLVMContext &getContext(jl_codectx_t &ctx) {
    return getContext(ctx.irbuilder);
}
```

Reply:
I think I'll make another refactoring PR at the very end of the globals deletion to clean up things like this (and maybe other small things like jl_Module becoming a function instead of a macro)